test/persistence.test.js

```diff
@@ -68,6 +90,27 @@ describe('Persistence', function () {
     assert.deepStrictEqual(treatedData[1], { _id: '3', nested: { today: now } })
   })
 
+  it('Badly formatted lines have no impact on the treated data (with stream)', function (done) {
+    const now = new Date()
+    const rawData = model.serialize({ _id: '1', a: 2, ages: [1, 5, 12] }) + '\n' +
+      'garbage\n' +
+      model.serialize({ _id: '3', nested: { today: now } })
+    const stream = new Readable()
+
+    stream.push(rawData)
+    stream.push(null)
+
+    d.persistence.treatRawStream(stream, function (err, result) {
+      console.log(err)
```

Comment:
No `console.log` in tests, please.
Should be replaced with an `assert` on the error (if we agree on the behavior it should have, then it'll be `assert.isNull(err)`)